src/web/releases.rs

```diff
@@ -244,57 +244,94 @@ fn get_releases_by_owner(
-                                    releases.release_time,
-                                    releases.rustdoc_status,
-                                    ts_rank_cd(crates.content, to_tsquery($1)) AS rank
-                                 FROM crates
-                                 INNER JOIN releases ON crates.latest_version_id = releases.id
-                                 WHERE crates.name LIKE concat('%', $1, '%')
-                                    OR crates.content @@ to_tsquery($1)
-                                 ORDER BY crates.name = $1 DESC,
-                                    crates.name LIKE concat('%', $1, '%') DESC,
-                                    rank DESC
-                                 LIMIT $2 OFFSET $3",
-        &[&query, &limit, &offset],
-    ) {
-        Ok(r) => r,
-        Err(_) => return None,
-    };
+    let rows = conn
+        .query(
+            "SELECT crates.name,
+                MAX(releases.version) AS version,
+                MAX(releases.description) AS description,
+                MAX(releases.target_name) AS target_name,
+                MAX(releases.release_time) AS release_time,
+                -- Cast the boolean into an integer and then cast it into a boolean.
+                -- Posgres moves in mysterious ways, don't question it
+                CAST(MAX(releases.rustdoc_status::integer) AS boolean) as rustdoc_status,
+                crates.github_stars,
+                SUM(releases.downloads) AS downloads,
+
+                -- The levenshtein distance between the search query and the crate's name
+                levenshtein_less_equal($1, crates.name, 3) as distance,
+                -- The similarity of the tokens of the search vs the tokens of `crates.content`.
+                -- The `32` normalizes the number by using `rank / (rank + 1)`
+                ts_rank_cd(crates.content, to_tsquery($2), 32)  as content_rank
+            FROM releases INNER JOIN crates on releases.crate_id = crates.id
+
+            -- Filter crates that haven't been built and crates that have been yanked
+            WHERE releases.rustdoc_status = true
+                AND releases.yanked = false
+                AND (
+                    -- Crates names that match the query sandwiched between wildcards will pass
+                    crates.name ILIKE CONCAT('%', $1, '%')
+                    -- Crate names with which the levenshtein distance is closer or equal to 3 will pass
+                    OR levenshtein_less_equal($1, crates.name, 3) <= 3
+                    -- Crates where their content matches the query will pass
+                    OR to_tsquery($2) @@ crates.content
+                )
+            GROUP BY crates.id
+            -- Ordering is prioritized by how closely the query matches the name, how closely the
+            -- query matches the description finally how many downloads the crate has
+            ORDER BY distance DESC,
```

Comment:
```suggestion
            -- NOTE: this means that exact matches will be shown first
            ORDER BY distance DESC,
```

Reply:
Shouldn't it be `ASC`? The lower the distance the closer to the query it is.